xontrib/uncpushd.py

```diff
@@ -0,0 +1,146 @@
+        co = subproc_captured_object(
+            ['reg', 'query', '"hkcu\software\microsoft\command processor"', '-v', 'DisableUNCCheck'])
+        if co.returncode == 0:
+            return 'enabled\n' if co.stdout[-3:-2] == '1' else 'disabled\n'\
+                , None, 0-int(co.stdout[-3:-2])
+        else:
+            return 'disabled\n', None, 0
+    else:                               # set to 1 or 0
+        co = subproc_captured_object(
+            ['reg', 'add', '"hkcu\software\microsoft\command processor"', '-v', 'DisableUNCCheck', '-t', 'REG_DWORD',
+             '-d', '1' if _uncpushd_choices[args.action] else '0', '-f'])
+        return None if co.returncode == 0 else co.stdout, None if co.returncode == 0 else co.stderr, co.returncode
+    pass
+
+import builtins
+import os
+from xonsh.dirstack import pushd, popd, DIRSTACK
+import subprocess
+
+def _do_subproc(args, msg=''):
+    """Because `subproc_captured_object` fails with error `Workstation Service not started` on a `NET USE dd: \\localhost\share`..."""
+    co = subprocess.run(args, universal_newlines=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    # if co.returncode != 0:
+    #     print('subproc returncode {}, msg: {}'.format(co.returncode, msg))
+    #     print(' '.join(co.args))
+    #     print(co.stdout)
+    #     print(co.stderr)
+    #     ##assert co.returncode==0, 'NET SHARE failed'
+    return co
+
+_unc_tempDrives = {}  # drivePart: tempDriveLetter for temp drive letters we create
+
+def unc_pushd( args, stdin=None):
+    """Handle pushd when argument is a UNC path. (\\<server>\<share>...)
+    If so, do like CMD.EXE pushd: create a temporary drive letter mapping, then pushd (via built-in) to that.
+    For this to work nicely, user must already have access to the UNC path
+    (e.g, via prior ```NET USE \\<server>\<share> /USER: ... /PASS:...```)
+
+    Author considers this behavior intrusive, so the extension does not create an alias for this function automatically.
+    If you want it, you must explicitly ask for it, e.g ```aliases['pushd']=unc_pushd; aliases['popd']=unc_popd```.
+    """
+    if not ON_WINDOWS or args is None or args[0] is None or args[0][0] not in (os.sep, os.altsep):
+        return pushd(args, stdin)
+    else:
+        share, relPath = os.path.splitdrive( args[0])
+        if share[0] not in (os.sep, os.altsep):
+            return pushd(args, stdin)
+        else:                                                # path begins \\ or //...
+            for dord in range(ord('z'), ord('a'), -1):
+                dpath = chr(dord) + ':'
+                if not os.path.isdir(dpath):                # find unused drive letter starting from z:
+                    co = _do_subproc(['net', 'use', dpath, share], msg='unc_pushd')
+                    if co.returncode != 0:
+                        return co.stdout, co.stderr, co.returncode
+                    else:
+                        _unc_tempDrives[dpath] = share
```

Comment:
PEP8 variables names should be `lower_case_with_underscores`, I see a lot of them like this here.